fmralign/_utils.py

```diff
@@ -49,85 +26,62 @@ def piecewise_transform(labels, estimators, X):
-    inverse_vals[vals] = np.arange(len(vals))
-    return inverse_vals[labels]
-
-
-def hierarchical_k_means(X, n_clusters):
-    """ Use a recursive k-means to cluster X
-    Parameters
-    ----------
-    X: nd array (n_samples, n_features)
-        Data to cluster
-    n_clusters: int
-        Number of clusters to output
-    Returns
-    -------
-    labels : list of ints (len n_features)
-        Parcellation of features in clusters
-    """
-
-    n_big_clusters = int(np.sqrt(n_clusters))
-    mbk = MiniBatchKMeans(init='k-means++', n_clusters=n_big_clusters, batch_size=1000,
-                          n_init=10, max_no_improvement=10, verbose=0,
-                          random_state=0).fit(X)
-    coarse_labels = mbk.labels_
-    fine_labels = np.zeros_like(coarse_labels)
-    q = 0
-    for i in range(n_big_clusters):
-        n_small_clusters = int(
-            n_clusters * np.sum(coarse_labels == i) * 1. / X.shape[0])
-        n_small_clusters = np.maximum(1, n_small_clusters)
-        mbk = MiniBatchKMeans(init='k-means++', n_clusters=n_small_clusters,
-                              batch_size=1000, n_init=10, max_no_improvement=10, verbose=0,
-                              random_state=0).fit(X[coarse_labels == i])
-        fine_labels[coarse_labels == i] = q + mbk.labels_
-        q += n_small_clusters
-
-    return _remove_empty_labels(fine_labels)
-
-
-def make_parcellation(X, mask, n_pieces, clustering_method='k_means', memory=Memory(cachedir=None), to_filename=None):
-    """Separates input data into pieces
+def _make_parcellation(imgs, clustering, n_pieces, masker, kmeans_smoothing_fwhm=5, verbose=0):
+    """Convenience function to use nilearn Parcellation class in our pipeline.
+    It is used to find local regions of the brain in which alignment will be later applied.
+    For alignment computational efficiency, regions should be of hundreds of voxels.
 
     Parameters
     ----------
-    X: numpy array of shape (n_samples, n_features)
+    imgs: Niimgs
         data to cluster
-    mask: numpy array
-        the mask from which X is produced
+    clustering: string or 3D Niimg
+        In : {'kmeans', 'ward', 'rena'}, passed to nilearn Parcellations class.
+        If you aim for speed, choose k-means (and check kmeans_smoothing_fwhm parameter)
+        If you want spatially connected and/or reproducible regions use 'ward'
+        For 'rena', need nilearn > 0.5.2
+        If 3D Niimg, image used as predefined clustering, n_pieces is ignored
     n_pieces: int
         number of different labels
-    clustering_method: string, optional
-        type of clustering applied to input data. Can be 'k_means', 'ward'
-    to_filename: str
-        path to which the parcellation will be saved
+    masker: instance of NiftiMasker or MultiNiftiMasker
+        Masker to be used on the data. For more information see:
+        http://nilearn.github.io/manipulating_images/masker_objects.html
+    kmeans_smoothing_fwhm: None or int
```

Comment:
smoothing_fwhm

Reply:
I really find it confusing to call that smoothing_fwhm as it is not the classical API, but one where a masker is provided. I really think we should keep the variable name explicit.

Reply:
No, i simply think that you need to explain in the docstring how it is used. I think that this becomes an issue is you also want to use `smoothing_fwhm` with the regular semantics of the parameter.